crawl-ref/source/cloud.cc

```diff
@@ -1814,3 +1816,94 @@ void end_still_winds()
     env.level_state &= ~LSTATE_STILL_WINDS;
     mpr("The air resumes its normal movements.");
 }
+
+
+void manage_cloud_generators() {
+    for (CloudGenerator &generator : env.cloud_generators)
+        generator.ran_this_turn = 0;
+    for (CloudGenerator &generator : env.cloud_generators)
+        generator.manage();
+}
+
+void CloudGenerator::manage()
```

Comment:
I had to change while (delay <= 0) to if (delay <= 0) for the chained loop, because it was infinite looping with ran_this_turn = 1 and delay > 0;

This leads me to believe I screwed up the loop. This function runs the cloud generators in order, unless they're part of a chained fog machine, in which case it will try to run all of them at the same time branching off of the first.

Expected behavior: what hangedman_minor_magic looks like right now. What actually happens: they just run like random cloud generators.

From what I can tell, normal cloud generators, assuming this code doesn't contain a memory leak, function as expected.

Reply:
"ran_this_turn" might be a mistake on my part due to not fully understanding the cloud code, but was my solution to running the cloud generators in the chained fog machine before/without knowledge of their own function call. This is probably the root of the problem